code/game/atoms/atom_movable/init_signals.dm

```diff
@@ -0,0 +1,192 @@
+	if(check_flags & PASS_AIR)
+		trait_list += TRAIT_PASS_AIR
+	if(check_flags & PASS_WALKOVER)
+		trait_list += TRAIT_PASS_WALKOVER
+	if(check_flags & PASS_TANK)
+		trait_list += TRAIT_PASS_TANK
+	return trait_list
+
+///Called on Initialize(), for the AM to register to relevant signals.
+/atom/movable/proc/register_init_signals()
+	RegisterSignal(src, SIGNAL_ADDTRAIT(TRAIT_PASS_LOW_STRUCTURE), PROC_REF(pass_low_structure_trait_gain))
+	RegisterSignal(src, SIGNAL_REMOVETRAIT(TRAIT_PASS_LOW_STRUCTURE), PROC_REF(pass_low_structure_trait_loss))
+
+	RegisterSignal(src, SIGNAL_ADDTRAIT(TRAIT_PASS_GLASS), PROC_REF(pass_glass_trait_gain))
+	RegisterSignal(src, SIGNAL_REMOVETRAIT(TRAIT_PASS_GLASS), PROC_REF(pass_glass_trait_loss))
+
+	RegisterSignal(src, SIGNAL_ADDTRAIT(TRAIT_PASS_GRILLE), PROC_REF(pass_grille_trait_gain))
+	RegisterSignal(src, SIGNAL_REMOVETRAIT(TRAIT_PASS_GRILLE), PROC_REF(pass_grille_trait_loss))
+
+	RegisterSignal(src, SIGNAL_ADDTRAIT(TRAIT_PASS_MOB), PROC_REF(pass_mob_trait_gain))
+	RegisterSignal(src, SIGNAL_REMOVETRAIT(TRAIT_PASS_MOB), PROC_REF(pass_mob_trait_loss))
+
+	RegisterSignal(src, SIGNAL_ADDTRAIT(TRAIT_PASS_DEFENSIVE_STRUCTURE), PROC_REF(pass_defensive_structure_trait_gain))
+	RegisterSignal(src, SIGNAL_REMOVETRAIT(TRAIT_PASS_DEFENSIVE_STRUCTURE), PROC_REF(pass_defensive_structure_trait_loss))
+
+	RegisterSignal(src, SIGNAL_ADDTRAIT(TRAIT_PASS_FIRE), PROC_REF(pass_fire_trait_gain))
+	RegisterSignal(src, SIGNAL_REMOVETRAIT(TRAIT_PASS_FIRE), PROC_REF(pass_fire_trait_loss))
+
+	RegisterSignal(src, SIGNAL_ADDTRAIT(TRAIT_PASS_XENO), PROC_REF(pass_xeno_trait_gain))
+	RegisterSignal(src, SIGNAL_REMOVETRAIT(TRAIT_PASS_XENO), PROC_REF(pass_xeno_trait_loss))
+
+	RegisterSignal(src, SIGNAL_ADDTRAIT(TRAIT_PASS_THROW), PROC_REF(pass_throw_trait_gain))
+	RegisterSignal(src, SIGNAL_REMOVETRAIT(TRAIT_PASS_THROW), PROC_REF(pass_throw_trait_loss))
+
+	RegisterSignal(src, SIGNAL_ADDTRAIT(TRAIT_PASS_PROJECTILE), PROC_REF(pass_projectile_trait_gain))
+	RegisterSignal(src, SIGNAL_REMOVETRAIT(TRAIT_PASS_PROJECTILE), PROC_REF(pass_projectile_trait_loss))
+
+	RegisterSignal(src, SIGNAL_ADDTRAIT(TRAIT_PASS_AIR), PROC_REF(pass_air_trait_gain))
+	RegisterSignal(src, SIGNAL_REMOVETRAIT(TRAIT_PASS_AIR), PROC_REF(pass_air_trait_loss))
+
+	RegisterSignal(src, SIGNAL_ADDTRAIT(TRAIT_PASS_WALKOVER), PROC_REF(pass_walkover_trait_gain))
+	RegisterSignal(src, SIGNAL_REMOVETRAIT(TRAIT_PASS_WALKOVER), PROC_REF(pass_walkover_trait_loss))
+
+	RegisterSignal(src, SIGNAL_ADDTRAIT(TRAIT_PASS_TANK), PROC_REF(trait_pass_tank_trait_gain))
+	RegisterSignal(src, SIGNAL_REMOVETRAIT(TRAIT_PASS_TANK), PROC_REF(trait_pass_tank_trait_loss))
+
+/obj/item/register_init_signals()
+	return //surely items should never need pass flags? Don't reg the signals as it can start to noticably effect load times
+
+/obj/effect/register_init_signals()
+	return //surely visual effects should never need pass flags? Don't reg the signals as it can start to noticably effect load times
```

Comment:
How does this pr compare to master init rn?

Reply:
right now its almost the same. without those overrides, it was adding like an extra 10sec or something.